### PR TITLE
Implement l dscaffold

### PR DESCRIPTION
**Pull request by gbstan92**
### JIRA link (if applicable) ###

AB#629

### Change description ###

Implemented ld-scaffold and updated azure-pipeline. Currently running with test data. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,13 @@
+provider "azurerm" {
+  version         = "=1.44.0"
+  subscription_id = "1baf5470-1c3e-40d3-a6f7-74bfbce4b348"
```
Comment (timja) [inline main.tf]: the subscription ID shouldn't be hardcoded here, it should go in the pipeline config

File: modules/ld-scaffold/roles.tf [hunk 2]
```diff
@@ -0,0 +1,40 @@
+resource "launchdarkly_custom_role" "roles" {
+  for_each = {
+    for project in var.project :
+    project.name => project
+  }
+
+  name = each.value.name
+  key  = each.value.azuread_group
+  //description = "Allows development access to users in the AzureAD group ${each.value.name} for the ${each.value.name} project"
```
Comment (timja) [inline modules/ld-scaffold/roles.tf]: don't leave commented out code in please, either fix or remove

File: terraform.tfvars [hunk 3]
```diff
@@ -0,0 +1,11 @@
+environments = [
+  "Production",
+  "AAT"
```
Comment (timja) [inline terraform.tfvars]: this is not correct, we should use the default environments that launch darkly provides, production and non prod I believe.

AAT is not what we've agreed so far as we want AAT to always have the same configuration as prod.

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+variable "projects" {
+  description = "Create projects with these names"
+}
+
+variable "tags" {
+  description = "Tags to be assigned to all resources created"
+  default = [
+    "terraform",
```
Comment (timja) [inline variables.tf]: is this useful to add?

File: README.md [hunk 5]
```diff
@@ -1,2 +1,21 @@
...
+A request to the DevOps team to create an Azure AD Group granting access to the LaunchDarkly application and include all users that will require access. You will require the Object ID of the Azure AD Group to complete the setup. 
+
+## Steps: 
+
+To add a new project, open the terraform.tfvars file and add the following:
+
+    - Name of the project
+    - Object ID of the Azure AD Group required in the prerequisites
+        
+
```
Comment (timja) [inline README.md]: Please remove all the blank lines below

File: variables.tf (hunk 4 above)
Comment (gbstan92) [reply variables.tf]: This shows us in the launchdarkly portal what resources are being created via terraform. It is also recommended by launchdarkly as best practice.

File: README.md (hunk 5 above)
Comment (gbstan92) [reply README.md]: Removed

File: terraform.tfvars (hunk 3 above)
Comment (gbstan92) [reply terraform.tfvars]: Switched back to default envs; Production and Test

**Pull request merged**